### PR TITLE
Social Media Website - Milestone 6: Friendships v2 

### PR DESCRIPTION
**Pull request by jamezjaz**
**On this branch, I did the following:**

- I implemented friendships with 2 rows for mutual friendship and
- I added more tests too.

### CODE REVIEW

**Review by euqueme — CHANGES_REQUESTED** [review]
> ### Pull Request reviewed by Maria Eugenia Queme P. (TSE)

### Hello @jamezjaz :lollipop: 

### Thank you for your submission :handshake: you've done a great job with this project :medal_sports: I like how you personalized the ALL USERS and Users profile views this makes your project stand out from the others :sparkles:  also all your associations are well made and working correctly in your models. let's work on the following changes :construction: :construction_worker: 

#### Behavior

- [ ] User can reject a friendship request ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/final_project/milestone5/accept_reject_friendship.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49736080/))

- users should be able to reject friendship request not just approve them so let's add this feature

#### No Logic in Views or partials

- [ ] Please use helper methods to handle complexity in views (like its explained in this article 
[The Beginner's Guide to Rails Helpers](https://mixandgo.com/learn/the-beginners-guide-to-rails-helpers)) there should be no if, if/else, unless or ternary operators within views or partials let's encapsulate this complex statements in helper methods, I found complexity in this view:

- app/views/posts/index.html.erb

#### General

- [ ] It seems like stickler is not activated and that's probably why the checkmarks are not appearing in your commits let's activate it [here](https://stickler-ci.com/)

#### [Observation]

- [ ] This pull request was requested only by you @jamezjaz but I see @MFahadUmer worked with you on the first milestone if this is a mistake and both of you worked in this milestone make sure that when you request for a re-review the " I work alone in this project" checkbox is unchecked.

### That's it! :raised_hands: 

### Please :pray: do the changes :technologist: and submit back for a code review when  you are done :memo: 

Happy coding! :rocket: 

Any questions? ask me on slack **@ Maru** :woman_technologist: 

PROJECT STATUS: REQUESTED CHANGES :vertical_traffic_light:

**Comment by jamezjaz** [comment]
As regards the [Observation], this is a collaborative project and as a result, my partner and I worked on previous milestones. However, Student Success had to unpair us because my partner changed his timezone and because of this, we were asked to remain solo until the end of Rails.
When I submitted for the last review, I left the "I work alone in this project" unchecked but I guess maybe the system does.
Thank you.

File: app/helpers/users_helper.rb [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  end
+
+  def friendship_status(user_id)
+    if current_user.friend?(user_id)
+      link_to 'Unfriend', friendship_path(user_id, method_name: 'delete'), method: :delete, class: 'friend_button'
+    elsif current_user.pend_friends.include?(user_id)
+      link_to 'Cancel', friendship_path(user_id, method_name: 'cancel'), method: :delete, class: 'friend_button'
+    elsif current_user.received_requests.include?(user_id)
+      # link_to 'Reject', friendship_path(user_id), method: :delete, class: 'friend_button'
+      # link_to 'Approve', friendship_path(user_id), method: :put, class: 'friend_button'
```
Comment (kobiyoyo) [inline app/helpers/users_helper.rb]: - [ ] Comments: It’s better to delete commented piece of codes. If you need to review a previous version of your code, use git history

File: app/controllers/friendships_controller.rb [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+      reverse_friendship.first.destroy
+      flash[:notice] = 'Friend Request Removed'
+    elsif params[:method_name] == 'cancel'
+      cancel_friendship = Friendship.where("user_id = #{current_user.id}"). where("friendship_id = #{params[:id]}")
+      cancel_friendship.first.destroy
+      flash[:notice] = 'Friend Request Cancelled'
+    else
+      friendship = Friendship.where("friendship_id = #{current_user.id}"). where("user_id = #{params[:id]}")
+      friendship.first.destroy
+      flash[:notice] = 'Friend Request Rejected'
```
Comment (kobiyoyo) [inline app/controllers/friendships_controller.rb]: Create a function for this and move it to friendship model

**Review by kobiyoyo — APPROVED** [review]
# Project Approved :thumbsup:
## Hello Team,  you have done a brilliant job with this project. :clap:
#### You can merge once you have implemented the changes below !!!

**Pull request merged**